src/strategy.rs

```diff
@@ -261,15 +271,400 @@ pub fn hostname<S: BuildHasher>(parameters: Option<HashMap<String, String, S>>)
     Box::new(move |_: &Context| -> bool { result })
 }
 
+/// returns true if the strategy should be delegated to, false to disable
+fn _compile_constraint_string<F>(expression: ConstraintExpression, getter: F) -> Evaluate
+where
+    F: Fn(&Context) -> Option<&String> + Clone + Sync + Send + 'static,
+{
+    match &expression {
+        ConstraintExpression::In(values) => {
+            let as_set: HashSet<String> = values.iter().cloned().collect();
```

Comment:
```suggestion
            let as_set = HashSet::from_iter(values);
```

Reply:
ta, shall do.

Reply:
Hmm, so actually - https://doc.rust-lang.org/std/iter/trait.FromIterator.html says 

> FromIterator's from_iter is rarely called explicitly, and is instead used through Iterator's collect method. See collect's documentation for more examples.

And this gives a good example why:

```
   Compiling unleash-api-client v0.4.0 (/home/robertc/src/unleash-client-rust)
error[E0282]: type annotations needed for `std::collections::HashSet<std::string::String, S>`
   --> src/strategy.rs:282:26
    |
282 |             let as_set = HashSet::from_iter(values.iter().cloned());
    |                 ------   ^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `S` declared on the struct `HashSet`
    |                 |
    |                 consider giving `as_set` the explicit type `std::collections::HashSet<std::string::String, S>`, where the type parameter `S` is specified

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
error: could not compile `unleash-api-client`.
```

S here is a RandomState type parameter. 

This:
```
   let as_set: HashSet<String> = HashSet::from_iter(values.iter().cloned());
```

Will compile successfully, but now we have stuttering and a longer line. We need the iter() and cloned() to get owned values, as &String has the wrong lifetime.

Consider the diff:
```
-            let as_set: HashSet<String> = values.iter().cloned().collect();
+            let as_set: HashSet<String> = HashSet::from_iter(values.iter().cloned());
```

Reply:
(I just realized that since the iterator is by reference this might not actually work the way I thought 😕 )

Reply:
Fair point :+1: